packages/password/src/index.mts

```diff
@@ -1,39 +1,110 @@
+  }
+
+  return '';
 };
 
-const password: Prompt<string, PasswordConfig> = (config, context) => {
+export default (config: PasswordConfig, context?: Context | undefined) => {
   if ('transformer' in config) {
     throw new Error(
       'Inquirer password prompt do not support custom transformer function. Use the input prompt instead.',
     );
   }
 
-  return input(
-    {
-      ...config, // Make sure we do not display the default password
-      default: undefined,
-      transformer(str: string, { isFinal }: { isFinal: boolean }) {
-        if (config.mask) {
-          const maskChar = typeof config.mask === 'string' ? config.mask : '*';
-          return maskChar.repeat(str.length);
-        }
+  return createPrompt<string, PasswordConfig>((_config, done) => {
+    const { validate = () => true, allowShowPassword } = _config;
+    const [status, setStatus] = useState<string>('pending');
+    const [errorMsg, setError] = useState<string | undefined>(undefined);
+    const [value, setValue] = useState<string>('');
+    const [togglePassword, setTogglePassword] = useState<boolean>(false);
 
-        if (!isFinal) {
-          return `${chalk.dim('[input is masked]')}${ansiEscapes.cursorHide}`;
+    const isLoading = status === 'loading';
+    const prefix = usePrefix(isLoading);
+
+    useKeypress(async (key, rl) => {
+      // Ignore keypress while our prompt is doing other processing.
+      if (status !== 'pending') {
+        return;
+      }
+
+      if (isEnterKey(key)) {
+        const answer = value;
+        setStatus('loading');
+        const isValid = await validate(answer);
+        if (isValid === true) {
+          setValue(answer);
+          setStatus('done');
+          done(answer);
+        } else {
+          // Reset the readline line value to the previous value. On line event, the value
+          // get cleared, forcing the user to re-enter the value instead of fixing it.
+          rl.write(value);
+          setError(isValid || 'You must provide a valid value');
+          setStatus('pending');
         }
+      } else if (allowShowPassword && key.ctrl === true && key.name === '`') {
+        // CTRL + Space
+        // I only tried on Linux, but the combination on Linux was reported like that
+        //   key.crtl = true
+        //   key.name = '`'
+        setTogglePassword(!togglePassword);
+        setValue(rl.line);
+        setError(undefined);
+      } else {
+        setValue(rl.line);
+        setError(undefined);
+      }
+    });
 
-        return '';
-      },
-    },
-    context,
-  );
-};
+    const message = chalk.bold(_config.message);
+    let formattedValue = transformer(value, _config.mask, { isFinal: status === 'done' });
+    if (togglePassword && status !== 'done') {
+      formattedValue = value;
+    }
 
-export default password;
+    if (status === 'done') {
+      formattedValue = chalk.cyan(formattedValue);
+    }
```

Comment:
I think there's a small logic mistake here. When the prompt is completed, we shouldn't show the password in plain text.

Reply:
When the prompt is completed, status will be equal to `'done'`, so the if at line 90 will be false, and `formattedValue` will remain equal to the transformed value. In the tests, I explicitly check for this scenario, in order to avoid password to be shown in plain text after the prompt is completed.